### DISCUSSION TITLE
winrt screen capture questions

### DISCUSSION THREAD

**Discussion by rlewkowicz** [obsproject/obs-studio#10023] [Development Questions]
winrt provides the most stable screen capture experience on windows. There's only a small handful of sample apps out there. In terms of implementation, obs has one of the nicer ones. 

I'm trying to put together something simple with cmake. The whole windows ecosystem is super convoluted to develop on imo. It's been a nightmare trying to manage packages to the point where I've pretty much thrown my hands up on it and will succumb to using visual studio just because the dependency management is more integrated. I know I don't have to build a uwp app though to make this work.

I hope someone can answer this though. How the heck does the winrt build work here? You're using methods as if you've imported header files, and done using namespace .... but there's not of that. I struggle to get half these calls to work across a dozen header files. There has to be a work flow and a converter? Or is who ever wrote this, just truly that brilliant, that they code exported c functions without meaningful compiler support? There had to be a functional prototype in c++ first? Or who ever did this just dropped right into some esoteric windows build environment and nailed it out of the gate?

**Comment by Fenrirthviti** [comment]
My guess without further info here: If you're looking at OBS code in a vacuum, you're probably missing all the parts where we pull things in from the Microsoft SDK, which is where a lot of the context would be.

**Comment by freedyalpha** [comment]
nice content
